iot-hub/Samples/service/IotHubSasCredentialAuthenticationSample/Program.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Azure.Core;
+using CommandLine;
+using Microsoft.Azure.Devices;
+using System;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IotHubSasCredentialAuthenticationSample
+{
+    public class Program
+    {
+        /// <summary>
+        /// A sample to illustrate how to use Azure active directory for authentication to the IoT hub.
+        /// <param name="args">Run with `--help` to see a list of required and optional parameters.</param>
+        public static async Task Main(string[] args)
+        {            
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // Initialize SAS token credentials.
+            Console.WriteLine("Creating sas credential.");
+
+            TimeSpan timeToLive = TimeSpan.FromHours(1);
+            DateTime expiresOn = DateTime.UtcNow.Add(timeToLive);
+            IotHubSasCredential sasCredential = new IotHubSasCredential(
+                GenerateSasToken(parameters.HostName, parameters.SharedAccessKey, expiresOn),
+                expiresOn);
+
+            // This is how the credential can be updated in the IotHubSasCredential object whenever necessary. It is not 
+            // necessary to do before the token expires. The following just demonstrates how to do it.
```

Comment:
would you like to understand what you meant to say by "It is not necessary to do before the token expires."

Reply:
sure. The SAS token expires. So update need not be called unnecessarily. In the sample, I'm calling it as soon as I created a token just to show the usage of the method.

Reply:
I can improve the wordings a bit.